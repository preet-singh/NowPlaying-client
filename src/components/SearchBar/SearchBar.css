.SearchBar input {
    border: none;
    border-radius: 5px;
    border-bottom: 1px solid grey;
    background-color: transparent;
    text-align: center;
    color: white;
}