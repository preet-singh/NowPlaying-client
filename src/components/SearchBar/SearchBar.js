import React from 'react'
import UserContext from '../../utils/context';
import {withRouter} from 'react-router-dom';
import './SearchBar.css'

class SearchBar extends React.Component {
  static contextType = UserContext;
  constructor(props) {
    super(props);
    this.state={

    }
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    let find = this.context.categoryItems.filter(item => item.title.toLowerCase().includes(this.state.title.toLowerCase())) || [];
    await this.context.setSearchedCategoryItems(find);
    this.props.history.push(`/category/${this.context.categoryID}`)
  }
  render() {
    return (
      <form className="SearchBar" onSubmit={(e) => this.handleSubmit(e)}>
        <input type="text" placeholder='ex. Frozen' id="searchBar" name="searchBar" aria-label="search bar" onChange={(e) => this.setState({title: e.target.value})} />
      </form>
    )
  }
}

export default withRouter(SearchBar);
