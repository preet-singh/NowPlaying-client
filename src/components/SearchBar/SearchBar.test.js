import React from 'react';
import SearchBar from './SearchBar.js';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`SearchBar component`, () => {

  it('renders the SearchBar', () => {
    const wrapper = shallow(<SearchBar />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})