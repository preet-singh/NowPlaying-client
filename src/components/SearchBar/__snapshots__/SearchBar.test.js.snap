// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders the SearchBar 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
